### PR TITLE
Improve comments and error messages around access tokens.

### LINKED ISSUE #10964: 'apparently some people use access tokens that are not macaroons'
This comment seems stale, since it now seems like we encourage and use `syt_...` tokens by default which are not macaroons?

https://github.com/matrix-org/synapse/blob/8c7a531e277f98ac6b7981b9738649f3a70feb94/synapse/api/auth.py#L417-L418

### PR DESCRIPTION
**Pull request by reivilibre**
Fixes #10964.

This commentary was confusing without the historical context (the comments are still roughly in a state where macaroons were the new thing, but now they're the old thing and opaque tokens are back in fashion!).
Especially notable was that 'Invalid macaroon passed.' is completely obscure to a client developer; customers have asked what this means!!

### CODE REVIEW

File: synapse/api/auth.py [hunk 1]
```diff
@@ -504,10 +512,9 @@ def _parse_and_validate_macaroon(
         try:
             macaroon = pymacaroons.Macaroon.deserialize(token)
         except Exception:  # deserialize can throw more-or-less anything
-            # doesn't look like a macaroon: treat it as an opaque token which
-            # must be in the database.
-            # TODO: it would be nice to get rid of this, but apparently some
-            # people use access tokens which aren't macaroons
+            # The access token doesn't look like a macaroon.
+            # In that case, we assume it's an opaque token which must be in the
+            # database.
```
Comment (clokep) [inline synapse/api/auth.py:517]: I think this gets called after we checked if it was in the database, so I'm not sure this is quite correct?

File: synapse/api/auth.py [hunk 2]
```diff
@@ -482,8 +484,14 @@ async def get_user_by_access_token(
             TypeError,
             ValueError,
         ) as e:
-            logger.warning("Invalid macaroon in auth: %s %s", type(e), e)
-            raise InvalidClientTokenError("Invalid macaroon passed.")
+            logger.warning(
+                "Invalid access token in auth: %s %s. (Neither a known token nor a valid macaroon.)",
```
Comment (clokep) [inline synapse/api/auth.py:488]: Would it be simpler to just say it is not a known access token and not mention macaroons at all?

```suggestion
                "Invalid access token in auth: %s %s.",
```

I'm OK either way, but figured I'd suggest it!

Comment (reivilibre) [reply synapse/api/auth.py]: Seems reasonable.

File: synapse/api/auth.py (hunk 1 above)
Comment (reivilibre) [reply synapse/api/auth.py:517]: I tried to write something and then wound up writing something which is easier to read in a different way. I'll try and rephrase.

Comment (reivilibre) [reply synapse/api/auth.py]: Maybe it's easier just to cut that sentence out.

File: synapse/api/auth.py [hunk 3]
```diff
@@ -482,8 +484,14 @@ async def get_user_by_access_token(
...
-            logger.warning("Invalid macaroon in auth: %s %s", type(e), e)
-            raise InvalidClientTokenError("Invalid macaroon passed.")
+            logger.warning(
+                "Invalid access token in auth: %s %s.",
+                type(e),
+                e,
+            )
+            raise InvalidClientTokenError(
+                "Invalid access token passed. (Neither a known token nor a valid macaroon.)"
+            )
```
Comment (clokep) [inline synapse/api/auth.py:494]:
```suggestion
            raise InvalidClientTokenError("Invalid access token passed.")
```

**Review by clokep — APPROVED** [review]
Assuming that lints pass. 👍

**Pull request merged**